.gitlab-ci.yml

```diff
@@ -230,3 +259,72 @@ coveralls:done:
     - curl "https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN&carryforward=frontend-unit,frontend-e2e,create-artifact-worker-unit,deployments-unit,deployments-integration,deviceauth-unit,deviceauth-integration,deviceconfig-unit,deviceconfig-integration,deviceconnect-unit,deviceconnect-integration,inventory-unit,inventory-integration,iot-manager-unit,iot-manager-integration,useradm-unit,useradm-integration,workflows-unit,workflows-integration" -d "payload[build_num]=$CI_PIPELINE_ID&payload[status]=done"
   tags:
     - hetzner-amd-beefy
+
+changelog:
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/node:20
+  stage: changelog
+  tags:
+    - hetzner-amd-beefy
+  rules:
+    - if: $RUN_RELEASE == "true"
+      when: never
+    - if: $CI_COMMIT_BRANCH =~ "/^v\d+\.\d+\.x$/"
+    - if: $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+  before_script:
+    # install release-please
+    - npm install -g release-please
+    # install github-cli
+    - |
+      mkdir -p -m 755 /etc/apt/keyrings \
+        && wget -qO- https://cli.github.com/packages/githubcli-archive-keyring.gpg | tee /etc/apt/keyrings/githubcli-archive-keyring.gpg > /dev/null \
+        && chmod go+r /etc/apt/keyrings/githubcli-archive-keyring.gpg \
+        && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
+        && apt update \
+        && apt install gh jq -y
+    # Setting up git
+    - |
+      echo "INFO - setting up git"
+      git config --global user.email "${GITHUB_USER_EMAIL}"
+      git config --global user.name "${GITHUB_USER_NAME}"
+    - npm install -g git-cliff
+    # GITHUB_TOKEN for Github cli authentication
+    - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+    # getting the centralized git cliff config
+    - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml.scoped
```

Comment:
Could we maybe maintain the `cliff.toml` as part of this repository?

Reply:
This file will be used in different repositories and having it in a shared location ensures the changelog will look consistent across our releases.

Reply:
> Could we maybe maintain the `cliff.toml` as part of this repository?

By having a central `cliff.toml`, we'll use a single "standard" for every projects, not just the mender-server, but also clients project. In the end, the central `cliff.toml` is what we want to achieve with: https://northerntech.atlassian.net/browse/QA-763